src/Symfony/Component/Console/Tests/Command/CommandTest.php

```diff
@@ -276,14 +276,42 @@ public function testRunNonInteractive()
         $this->assertEquals('execute called'.PHP_EOL, $tester->getDisplay(), '->run() does not call the interact() method if the input is not interactive');
     }
 
-    public function testExecuteMethodNeedsToBeOverridden()
+    public function testExecuteMethodThrowsExceptionWhenNotOverridden()
     {
         $this->expectException('LogicException');
         $this->expectExceptionMessage('You must override the execute() method in the concrete command class.');
         $command = new Command('foo');
         $command->run(new StringInput(''), new NullOutput());
     }
 
+    public function testExecuteMethodThrowsExceptionWhenExtendedButNotOverridden()
+    {
+        $childCommand = new class() extends Command {
+        };
+
+        $this->expectException('LogicException');
+        $this->expectExceptionMessage('You must override the execute() method in the concrete command class.');
```

Comment:
without your change, this test still passes isnt it? not sure what it solves currently :S

Reply:
@ro0NL The current code of `Command::execute()` is not compatible with the declared return type `int` because it always throws an exception and returns nothing (see linked commit in description). That is what this PR is about.

My change keeps the functionality of the expected exception intact (in the context of the test method on which you commented), which in my view is just backwards compatibility.

In addition to this, I've added testing for the return type, as it was already specified in the docblock.

Reply:
why would it be incompatible? https://3v4l.org/hKjmS

Reply:
Because an thrown exception is not an int.

If you explicitly declare `* @return int 0 if everything went fine, or an exit code` and the only line in the method is a thrown exception, how can the return value ever be compatible with the documented return type?

Reply:
because throwing aborts the code execution, shifting to the first `catch` block somewhere else.

Currently it's equivalent with:

```
throw new Exception(...);
return 0;
```

but as throw aborts code execution, the `return` is unreachable ... so safe to remove. BUT if this method was ever to return something, then it must be int.

Im :-1: for this PR. Personally i think it makes things more confusing...

Reply:
if anything, it could be

```
{
   throw new LogicException('You must override the execute() method in the concrete command class.');

   return 0;
}
```

but it's as confusing as the current conditional approach IMHO.

Reply:
If I understand the code correctly, this would change the behavior when calling `parent::execute(...)`. Previously this would throw an exception that you'd have to implement the method, which makes no sense as you've implemented it. In the new situation it would return `0` if you have implemented the method when calling the parent.

I'm not sure if it makes sense to call the parent execute method though.